modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/Common.cs

```diff
@@ -67,6 +67,28 @@ TypeDeclarationSyntax outerTypeDeclSyntax
                 outerTypeDeclSyntax.SyntaxTree.FilePath));
         }
 
+        public static void ReportMultipleClassesInGodotScript(
+            GeneratorExecutionContext context,
+            ClassDeclarationSyntax cds, INamedTypeSymbol symbol
+        )
+        {
+            string message = $"Found multiple classes with the name '{symbol.Name}' in the same script file";
+
+            string description = $"{message}. A script file must only contain one class with a name that matches the file name.";
+
+            context.ReportDiagnostic(Diagnostic.Create(
+                new DiagnosticDescriptor(id: "GD0003",
```

Comment:
Can we keep this in a `static readonly`, I'd love for all the diagnostics to slowly be written in the same way. E.g. like `GenericTypeArgumentMustBeVariantRule`.

Reply:
I'm inclined to leave this for a future PR and just follow the current established pattern for now. But I agree with your suggestion and we should definitely do that at some point.

Reply:
That is fine either way for me.